### PR TITLE
Handle repository deletions

### LINKED ISSUE #301: Disable subscriptions when a repository is deleted
Possibly the cause for #300

### PR DESCRIPTION
**Pull request by wilhelmklopp**
At the moment when a repository gets deleted we don't do anything about it.

With this PR, when a repository gets deleted, we post a message to the relevant channels (example below) and delete the subscription:
image

Closes #301

### CODE REVIEW

File: lib/activity/router.js [hunk 1]
```diff
@@ -27,6 +27,13 @@ module.exports = ({ models }) => {
             include: [GitHubUser],
           });
 
+          const eventType = `${context.event}.${context.payload.action}`;
+
+          // Can't check for repo access because repo has been deleted
+          if (eventType === 'repository.deleted') {
+            return callback(context, subscription, slack);
+          }
+
```
Comment (wilhelmklopp) [inline lib/activity/router.js]: @bkeepers I wonder if there's a better way to do this

File: lib/models/subscription.js [hunk 2]
```diff
@@ -24,6 +24,10 @@ module.exports = (sequelize, DataTypes) => {
     reviews: false,
   };
 
+  const FixedSettings = {
```
Comment (wilhelmklopp) [inline lib/models/subscription.js]: This should probably have a better name - and maybe not even include the word 'settings'?

Comment (bkeepers) [reply lib/models/subscription.js]: Hmm, should this be the same setting as `public` events? It looks like they're included in the same `repository` webhook: https://developer.github.com/v3/activity/events/types/#repositoryevent.

Comment (wilhelmklopp) [reply lib/models/subscription.js]: the `public` event actually seems to come in via two webhooks. One as part of the repository event and one as part of the public event: https://developer.github.com/v3/activity/events/types/#publicevent

Right now we listen to the public event

File: lib/messages/repository-deleted.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  toJSON() {
+    const pretext = `Repository deleted by ${this.sender.login}`;
+
+    return {
+      attachments: [{
+        ...super.getBaseMessage(),
+        fallback: pretext,
+        pretext,
+        text: 'The subscription to this repository has been disabled.',
```
[the remark this answers is not in the document]
Comment (wilhelmklopp) [reply lib/messages/repository-deleted.js]: Yeah, that's a really good point. 👍

**Pull request merged**